src/benchmarklib/benchmark_sql_executor.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "sql/sql_pipeline.hpp"
+#include "utils/sqlite_wrapper.hpp"
+
+namespace opossum {
+
+class TransactionContext;
+
+// This class provides SQL functionality to BenchmarkItemRunners. See AbstractBenchmarkItemRunner:_execute_item.
+class BenchmarkSQLExecutor {
+ public:
+  // If sqlite_wrapper is set, all SQL queries executed in the benchmark will also be executed using SQLite and then
+  // validated.
+  BenchmarkSQLExecutor(bool enable_jit, const std::shared_ptr<SQLiteWrapper>& sqlite_wrapper,
+                       const std::optional<std::string>& visualize_prefix);
+
+  std::shared_ptr<const Table> execute(const std::string& sql);
+  std::vector<SQLPipelineMetrics> metrics;
```

Comment:
pls doc: one entry for each execute() call, i assume?
 also a newline before this would lead to better grouping of members